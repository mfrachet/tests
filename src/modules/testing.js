import { render as renderTl } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import { QueryClient, QueryClientProvider } from "react-query";
import { IntlProvider } from "react-intl";
import { ThemeProvider } from "styled-components";
import { theme } from "../theme";
import en from "../translations/en.json";

export const render = (component) => {
  const queryClient = new QueryClient();
  return renderTl(
    <QueryClientProvider client={queryClient}>
      <IntlProvider locale={navigator.language || "en"} messages={en}>
        <ThemeProvider theme={theme}>
          <MemoryRouter>{component}</MemoryRouter>
        </ThemeProvider>
      </IntlProvider>
    </QueryClientProvider>
  );
};
