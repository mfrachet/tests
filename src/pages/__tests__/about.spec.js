import { render } from "../../modules/testing";
import About from "../about";

describe("About", () => {
  it("does something great", () => {
    const { container } = render(<About />);

    expect(container).toMatchSnapshot();
  });
});
