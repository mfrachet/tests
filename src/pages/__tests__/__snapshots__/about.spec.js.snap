// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About does something great 1`] = `
<div>
  <div>
    About
  </div>
</div>
`;
